### PR TITLE
Add broadcasting for dot/vdot

### PR DESCRIPTION
**Pull request by heitorschueroff**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47213 Add broadcasting for dot/vdot**

Task from here #https://github.com/pytorch/pytorch/issues/42666

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -2563,31 +2563,40 @@ def merge_dicts(*dicts):
...
            r"""
 vdot(input, other, *, out=None) -> Tensor
 
-Computes the dot product (inner product) of two tensors. The vdot(a, b) function
+Computes the dot product (inner product) of two 1D tensors. The vdot(a, b) function
 handles complex numbers differently than dot(a, b). If the first argument is complex,
 the complex conjugate of the first argument is used for the calculation of the dot product.
 
-.. note:: This function does not :ref:`broadcast <broadcasting-semantics>`.
+.. note:: This function is similar to numpy.vdot only for 1D tensors.
```
Comment (mruberry) [inline torch/_torch_docs.py:2595]: .. note:: Unlike NumPy's `vdot`, `torch.vdot` intentionally only supports computing the dot product of two 1D tensors with the same number of elements.

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -2602,6 +2611,8 @@ def merge_dicts(*dicts):
     tensor([16.+1.j])
     >>> torch.vdot(b, a)
     tensor([16.-1.j])
+    >>> torch.vdot(a, c)
```
Comment (mruberry) [inline torch/_torch_docs.py:2614]: Where does `c` come from in this example?

File: torch/_torch_docs.py (hunk 1 above)
Comment (mruberry) [reply torch/_torch_docs.py:2595]: Same wording suggestion for the previous documentation, too.

File: test/test_torch.py [hunk 3]
```diff
@@ -17021,6 +17021,13 @@ def compare_out_variant(torch_fn, x, y):
             compare_with_numpy_bin_op(torch_fn, np_fn, v4, v1)
             compare_out_variant(torch_fn, v4, v1)
 
+        # Test broadcasting
```
Comment (mruberry) [inline test/test_torch.py:17024]: Let's move the dot tests to test_linalg.py.

Is the out variant tested?

**Comment by mruberry** [comment]
Summary of offline discussion: my mistake on the "broadcasting" part of this task. We don't want to support broadcasting in dot or vdot. I suggested a docs change that should clarify the intentionality of this decision.

**Comment by heitorschueroff** [comment]
Closing this PR. New PR that only updates docs/test here https://github.com/pytorch/pytorch/pull/47242

**Pull request closed without merging**